compiler/safetylib/safetyArch.ml

```diff
@@ -0,0 +1,48 @@
+open Jasmin
```

Comment:
`pd` and `asmOp` should be in the functor rather than function arguments. Apart from the that, the signature is convincing.